extensions/wikia/PortableInfobox/tests/XmlParserTest.php

```diff
@@ -91,4 +91,21 @@ public function errorHandlingDataProvider() {
 			]
 		];
 	}
+
+	/**
+	 * @dataProvider entitiesTestDataProvider
+	 */
+	public function testHTMLEntities( $markup, $expectedResult ) {
+		$result = \Wikia\PortableInfobox\Parser\XmlParser::parseXmlString( $markup );
+		$this->assertEquals( $expectedResult, $result[ 0 ] );
+	}
+
+	public function entitiesTestDataProvider() {
```

Comment:
You might consider adding a few test cases for entities like
&Delta; 
&Uuml;
and coumpound cases, i.e.
&amp;amp;
or malformed entities like
&amp
&middot